src/common/database.c

```diff
@@ -1415,6 +1415,33 @@ static int _upgrade_library_schema_step(dt_database_t *db, int version)
 
     new_version = 23;
   }
+  else if(version == 23)
+  {
+    sqlite3_exec(db->handle, "BEGIN TRANSACTION", NULL, NULL, NULL);
+    TRY_EXEC("CREATE TABLE main.history_hash (imgid INTEGER PRIMARY KEY, "
+             "initial_hash BLOB, default_hash BLOB, current_hash BLOB)",
+             "[init] can't create table history_hash\n");
+
+    const gsize checksum_len = g_checksum_type_get_length(G_CHECKSUM_MD5);
+    guint8 *checksum = calloc(1, checksum_len);
+    sqlite3_stmt *ins_stmt = NULL;
+    char *query = dt_util_dstrcat(NULL, "INSERT INTO main.history_hash "
```

Comment:
The LEFT JOIN may look surprising, I would add a comment:

// Insert an history_hash entry for all images with some history already set (skip non edited images)

Reply:
The left join is to get also images without history.

// Insert an history_hash entry for all images with some history already set (including  non edited images)

Reply:
done